Makefile

```diff
@@ -519,7 +522,28 @@ release-binary: $(RELEASE_DIR)
 release-staging: ## Builds and push container images to the staging bucket.
 	REGISTRY=$(STAGING_REGISTRY) $(MAKE) docker-build-all docker-push-all release-alias-tag
 
-RELEASE_ALIAS_TAG=$(PULL_BASE_REF)
+.PHONY: release-staging-daily
+release-staging-daily: ## Builds and push container images to the staging bucket.
+	$(eval RELEASE_ALIAS_TAG := daily_$(TAG)_$(shell date +%F))
+	$(MAKE) release-staging RELEASE_ALIAS_TAG=$(RELEASE_ALIAS_TAG)
+
+	# Set the images to the staging bucket.
+	$(MAKE) set-manifest-image \
+		MANIFEST_IMG=$(STAGING_REGISTRY)/$(IMAGE_NAME) MANIFEST_TAG=$(RELEASE_ALIAS_TAG) \
+		TARGET_RESOURCE="./config/manager/manager_image_patch.yaml"
+	$(MAKE) set-manifest-image \
+		MANIFEST_IMG=$(STAGING_REGISTRY)/$(KUBEADM_BOOTSTRAP_IMAGE_NAME) MANIFEST_TAG=$(RELEASE_ALIAS_TAG) \
+		TARGET_RESOURCE="./bootstrap/kubeadm/config/manager/manager_image_patch.yaml"
+	$(MAKE) set-manifest-image \
+		MANIFEST_IMG=$(STAGING_REGISTRY)/$(KUBEADM_CONTROL_PLANE_IMAGE_NAME) MANIFEST_TAG=$(RELEASE_ALIAS_TAG) \
+		TARGET_RESOURCE="./controlplane/kubeadm/config/manager/manager_image_patch.yaml"
+	# Set pull policy
+	$(MAKE) set-manifest-pull-policy PULL_POLICY=IfNotPresent TARGET_RESOURCE="./config/manager/manager_pull_policy.yaml"
+	$(MAKE) set-manifest-pull-policy PULL_POLICY=IfNotPresent TARGET_RESOURCE="./bootstrap/kubeadm/config/manager/manager_pull_policy.yaml"
+	$(MAKE) set-manifest-pull-policy PULL_POLICY=IfNotPresent TARGET_RESOURCE="./controlplane/kubeadm/config/manager/manager_pull_policy.yaml"
+	## Build the manifests
+	$(MAKE) release-manifests
+	gsutil cp $(RELEASE_DIR)/* gs://$(STAGING_BUCKET)/components/$(RELEASE_ALIAS_TAG)
```

Comment:
This seems a duplicate of what we have in the release target. Have you considered refactoring to avoid such duplication?

Also, I think we should add release-manifests-dev and make sure CAPD is part of the release package

Reply:
Do we need CAPD images and manifests in the daily release though? CAPI e2e won't be using those.

Reply:
CAPD is part of the release bucket, is part of the build that is triggered at every commit, so I'm +1 to be consistent in the daily bucket as well

Reply:
Removed duplication and added CAPD.